commitment_test.go

```diff
@@ -0,0 +1,159 @@
+package raft
+
+import (
+	"testing"
+)
+
+// Returns a slice of server names of size n.
+func voters(n int) []string {
+	if n > 7 {
+		panic("only up to 7 servers implemented")
+	}
+	return []string{"s1", "s2", "s3", "s4", "s5", "s6", "s7"}[:n+1]
+}
+
+// Tests setVoters() keeps matchIndexes where possible.
+func TestCommitment_setVoters(t *testing.T) {
+	commitCh := make(chan struct{}, 1)
```

Comment:
Would be good in these tests to assert that the `commitCh` got hit correctly (or at least the correct number of times).